app/models/topic.rb

```diff
@@ -1040,7 +1040,9 @@ def invite(invited_by, username_or_email, group_ids = nil, custom_message = nil)
         raise UserExists.new(I18n.t("topic_invite.user_exists"))
       end
 
-      ensure_can_invite!(target_user, invited_by)
+      communication_preferences = UserCommunicationDefender.new(acting_user_id: invited_by.id, target_usernames: target_user.username).fetch_user_preferences
+      target_user_communication_preferences = communication_preferences.for_user(target_user.id)
+      raise NotAllowed.new(I18n.t("not_accepting_pms", username: target_user.username)) if !communication_preferences.acting_user_staff? && target_user_communication_preferences&.ignoring_or_muting?
```

Comment:
I personally feel like the new API is way more verbose than the one liner previously. Perhaps `UserCommunicationDefender` can have a `#can_invite_user` method that returns a boolean? 

In some way, I feel like the user preferences should be encapsulated within the `UserCommunicationDefender` itself.

Reply:
Ohh yes that would be better, that way it can be a lot shorter like this:

```ruby
communication_preferences = UserCommsScreener.new(acting_user_id: invited_by.id, target_usernames: target_user.username)
if !communication_preferences.can_invite_user?(target_user.id)
  raise NotAllowed.new(I18n.t("not_accepting_pms", username: target_user.username))
end
```

We can dispense with the need to call `fetch_user_preferences` and just do that lazily as required, and maybe just add another method called `preferences` that gives you that raw data if you do need it.

Reply:
Yup :) I like the shorter version better.